jax/_src/numpy/reductions.py

```diff
@@ -782,9 +781,59 @@ def average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None, *
 @overload
 def average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
             returned: bool = False, keepdims: bool = False) -> Array | tuple[Array, Array]: ...
-@implements(np.average)
 def average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
             returned: bool = False, keepdims: bool = False) -> Array | tuple[Array, Array]:
+  """Compute the weighed average.
+
+  JAX Implementation of :func:`numpy.average`.
+
+  Args:
+    a: array to be averaged
+    axis: optional, int or sequence of ints, default=None. Axis along which
+      the mean to be computed. If None, mean is computed along all the axes.
+    weights: an optional array of weights for a weighted average. Must be
+      broadcast-compatible with ``a``.
+    returned: bool, default=False. If true then return the normalization factor
+      (i.e. the sum of weights).
+    keepdims: bool, default=False. If true, reduced axes are left in the result
```

Comment:
A question for future PRs: do you think we should drop types/defaults from docstrings, since they are available in the function signature anyway?

Reply:
Yeah in general I think you're right. I copied these from the `jnp.mean` docstring, but I'll change them for this PR